### ISSUE TITLE
Natural gradient for neural network?

### ISSUE THREAD

**Issue by JiahaoYao** [wiseodd/natural-gradients#3]
Hi @wiseodd, thanks for great implementation! I wonder whether this can be generalized to a simple feed-forward network. In your code, [the gradient wrt the parameters](https://github.com/wiseodd/natural-gradients/blob/master/numpy/toy/full_fisher.py#L47) are computed directly. For networks, can you use auto-differentiation for that, say in pytorch? I find that is difficult because you need to get the gradient of the size (N, num_of_thetas). Since in my knowledge, when one does backpropagation, the loss should be a scalar, thus the batch N is omitted. What can one do to get the gradient for each instance in the batch?
